frontend/src/pages/DegreeWizard/DegreeWizard.jsx

```diff
@@ -108,34 +112,23 @@ const DegreeWizard = () => {
         <hr className="rule" />
 
         <div className="steps-container">
-          {currStep >= 1 && (
-            <div className="step-content" id="Year">
-              <YearStep incrementStep={incrementStep} currStep={currStep} />
-            </div>
-          )}
-          {currStep >= 2 && (
-            <div className="step-content" id="Degree">
-              <DegreeStep incrementStep={incrementStep} currStep={currStep} />
-            </div>
-          )}
-          {currStep >= 3 && (
-            <div className="step-content" id="Specialisation">
-              <SpecialisationStep
-                incrementStep={incrementStep}
-                currStep={currStep}
-              />
-            </div>
-          )}
-          {currStep >= 4 && (
-            <div className="step-content" id="Minor">
-              <MinorStep incrementStep={incrementStep} currStep={currStep} />
-            </div>
-          )}
-          {currStep >= 5 && (
-            <div className="step-content" id="Start Browsing">
-              <StartBrowsingStep />
-            </div>
-          )}
+          {stepList.map((stepName, index) => (
+            currStep >= index + 1 && (
+              <div className="step-content" id={stepName}>
+                {
+                  index + 1 === 1 ? <YearStep currStep={currStep} incrementStep={incrementStep} />
+                    : index + 1 === 2 ? (
+                      <DegreeStep
+                        currStep={currStep}
+                        incrementStep={incrementStep}
+                      />
+                    )
+                      : index + 1 === stepList.length ? <StartBrowsingStep />
+                        : <SpecialisationStep incrementStep={incrementStep} type={stepName} />
+                }
+              </div>
+            )
+          ))}
```

Comment:
this can be less ugly somehow i think - does an FE person have any better ideas?

Reply:
Perhaps could just extract this into a generic `Step` component where you can use regular `if` statements to conditionally render instead of doing wacky ternary conditions inside the jsx. I suppose hussain's suggestion of precomputation could also work, with something like a `steps[index] ?? <SpecialisationStep ...`

Personally I would also use `.slice` before `map` here instead of having that `currStep >= index + 1` condition adding an extra layer of complexity (btw actually took me a while to realise you were rendering the first n, not the last n, because of the `>=` confusing me - `index` being on the right is strange imo but that's just my personal style preference)

Also should probably be giving the divs here a `key` prop? :p

Reply:
Couldn't you have it where you use `map` on `steps` rather than `stepList`? Whilst also keep the old way of rendering the year, degree and start browsing step (by hard coding it)?

Something like this:
```
{steps.map((stepName, index) => (
    currStep >= index + 3 && (
    <div className="step-content" id={stepName}>
      <SpecialisationStep incrementStep={incrementStep} type={stepName} />
    </div>
    )
  ))}
```
May not be the best approach but i feel its more readable than nested ternary statements 😅